internal/util/update/update.go

```diff
@@ -207,35 +215,41 @@ func (nrc *nilRepoClone) AbsPath() string {
 // updateRootPackage updates a local package. It will use the information
 // about upstream in the Kptfile to fetch upstream and origin, and then
 // recursively traverse the hierarchy to add/update/delete packages.
-func (u Command) updateRootPackage(p *pkg.Pkg) error {
+func (u Command) updateRootPackage(ctx context.Context, p *pkg.Pkg) error {
+	const op errors.Op = "update.updateRootPackage"
 	kf, err := p.Kptfile()
 	if err != nil {
-		return err
+		return errors.E(op, p.UniquePath, err)
 	}
 
 	if kf.Upstream == nil || kf.Upstream.Git == nil {
 		return nil
 	}
 
 	g := kf.Upstream.Git
+
+	prntr := printer.FromContextOrDie(ctx)
```

Comment:
we should probably standardize on the local variable name for printer. I used `pr` but wasn't super happy about it, but found to be better than alternatives.